elara/elarautil.py

```diff
@@ -4,62 +4,108 @@
-                "Store JSON error. File might be encrypted. Run in secure mode with key path."
-            )
+    def check_encrypted(version):
+        # if msb of version number is set the db is encrypted
+        return (version & (1 << 15)) != 0
+
+    @staticmethod
+    def read_plain_db(obj) -> Dict:
+        with open(obj.path, "rb") as fctx:
+            if not Util.check_mag(fctx.read(4)):
+                raise FileAccessError("File magic number not known")
+            version = int.from_bytes(fctx.read(2), "little", signed=False)
+            # check for encryption before trying anything
+            if Util.check_encrypted(version):
+                raise FileAccessError("This file is encrypted, run in secure mode")
+            checksum = int.from_bytes(fctx.read(4), "little", signed=False)
+            data = fctx.read()
+            calculated_checksum = crc32(data)
+            if calculated_checksum != checksum:
+                raise LoadChecksumError(
+                    f"calculated checksum: {calculated_checksum} is different from stored checksum {checksum}")
+            elif version != obj.db_format_version:
+                raise LoadIncompatibleDB(f"db format version {version} is incompatible with {obj.db_format_version}")
+            try:
+                curr_db = msgpack.unpackb(data)
+            except FileNotFoundError:
+                raise FileAccessError(
+                    "File not found"
+                )
+            return curr_db
+
+    @staticmethod
+    def store_plain_db(obj):
+        with open(obj.path, "wb") as fctx:
+            try:
+                data = msgpack.packb(obj.db)
+                buffer = b"ELDB"
+                buffer += obj.db_format_version.to_bytes(2, "little")
+                buffer += (crc32(data)).to_bytes(4, "little")
+                buffer += data
+                fctx.write(buffer)
+            except FileExistsError:
+                raise FileAccessError(
+                    "File already exists"
+                )
 
     @staticmethod
-    def readAndDecrypt(obj):
+    def read_and_decrypt(obj):
         if obj.key:
             fernet = Fernet(obj.key)
-            encrypted_data = None
             try:
-                with open(obj.path, "rb") as file:
-                    encrypted_data = file.read()
-            except Exception:
+                with open(obj.path, "rb") as fctx:
+                    if not Util.check_mag(fctx.read(4)):
+                        raise FileAccessError("File magic number not known")
+                    version = int.from_bytes(fctx.read(2), "little")
+                    if not Util.check_encrypted(version):
+                        raise FileAccessError("File is marked not encrypted, you might have a corrupt db")
+                    checksum = int.from_bytes(fctx.read(4), "little")
+                    encrypted_data = fctx.read()
+                    calculated_checksum = crc32(encrypted_data) & 0xFFFFFFFF
```

Comment:
What exactly is the role of `0xFFFFFFFF` here?

Reply:
oops seems like I missed one, it should not be there 😛

Reply:
crc32 used to return an integer not not necessarily an unsigned one force of habit

Reply:
@DarthUdp Alright, can you please update that and issue another pull request?

Reply:
there is no need to issue another pr github will conciliate the two commits

Reply:
Done :)